### PR TITLE
DM-37478: migrating to model packages

### PR DESCRIPTION
**Pull request by NimSed**
Migrating to model packages. Main changes have been applied to the meas_transiNet repository, and this pull request only contains minor necessary changes to keep the two repositories compatible.

### CODE REVIEW

File: rbResnet50-DC2/rbResnet50.py [hunk 1]
```diff
@@ -16,7 +16,7 @@ class RBResNet50(nn.Module):
     def __init__(self):
         super().__init__()
 
-        self.resnet = resnet50(pretrained=False)
+        self.resnet = resnet50(weights=None)
```
Comment (kfindeisen) [inline rbResnet50-DC2/rbResnet50.py:19]: Is this happening alongside an update to our version of `torchvision`, or is this a past incompatibility that we missed for lack of testing?

File: rbResnet50-DC2/rbResnet50.py [hunk 2]
```diff
@@ -16,7 +16,7 @@ class RBResNet50(nn.Module):
     def __init__(self):
```
Comment (kfindeisen) [inline rbResnet50-DC2/rbResnet50.py:16]: Can the DOI link in the comment above be replaced with a published Microsoft(?) report?

File: model_packages/rbResnet50-DC2/test.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+
+class TestModelPackageNeighbor(unittest.TestCase):
+    def setUp(self):
+        self.model_package_name = 'neighbor:///rbResnet50-DC2'
+
+    def test_load(self):
+        """Test loading of a 'neighbor' model package
+        """
+        model_package = NNModelPackage(self.model_package_name)
```
Comment (kfindeisen) [inline model_packages/rbResnet50-DC2/test.py]: The URI-based API for `NNModelPackage` is out of date, please update accordingly and verify the tests.

Comment (kfindeisen) [reply model_packages/rbResnet50-DC2/test.py]: How is this test run, anyway? I see that `rbClassifier_data` is buildable with Scons, but this organization of tests won't work with the default LSST tooling. If you can't set up a custom Scons runner for these tests, then moving them into per-model files in a `tests/` directory is probably better than not running them at all.

File: model_packages/rbResnet50-DC2/test.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        self.interface = RBTransiNetInterface(model_package_name)
+
+    def test_infer_empty(self):
+        """Test running infer on images containing all zeros.
+        """
+        data = np.zeros((256, 256), dtype=np.single)
+        inputs = CutoutInputs(science=data, difference=data, template=data)
+        result = self.interface.infer([inputs])
+        self.assertTupleEqual(result.shape, (1,))
+        self.assertAlmostEqual(result[0], 0.00042127)  # Empirical meaningless value spit by this very model
```
Comment (kfindeisen) [inline model_packages/rbResnet50-DC2/test.py:67]: Maybe reword the comment to "Test for consistency with past behavior, given the same inputs and weights"?

File: model_packages/rbResnet50-DC2/test.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        self.assertTupleEqual(weights.shape, (64, 3, 7, 7))
+
+        # test weight values
+        torch.testing.assert_close(weights[0][0][0],
+                                   torch.tensor([-0.06167374, 0.0780527, 0.01648014, -0.05989857,
+                                                 0.01510609, 0.06968862, 0.05509452]),
+                                   rtol=1e-8, atol=1e-8)
+
+
+class TestOneCutout(unittest.TestCase):
```
Comment (kfindeisen) [inline model_packages/rbResnet50-DC2/test.py]: As in `meas_transiNet`, why have a separate class for this test? The current arrangement requires duplicating the setup code for each test case, which defeats the point of having a `SetUp`.

**Review by kfindeisen — CHANGES_REQUESTED** [review]
Most of the changes look fine, but I'm concerned that the tests don't correspond to the latest API on lsst/meas_transiNet#12. Do they really pass in their current form?

File: rbResnet50-DC2/rbResnet50.py (hunk 1 above)
Comment (NimSed) [reply rbResnet50-DC2/rbResnet50.py:19]: Yes, the former.

File: model_packages/rbResnet50-DC2/test.py (hunk 3 above)
Comment (NimSed) [reply model_packages/rbResnet50-DC2/test.py]: This is the resurrected test, being stored there for potential future reference -- i.e. is not currently being called/run by anyone.
But I'm going to apply the suggested fixes and make it runnable.

File: model_packages/rbResnet50-DC2/test.py (hunk 4 above)
Comment (NimSed) [reply model_packages/rbResnet50-DC2/test.py:67]: I see this comment's purpose more as an answer to some one who asks what is "0.00042127". Let's leave it?

File: model_packages/rbResnet50-DC2/test.py (hunk 5 above)
Comment (NimSed) [reply model_packages/rbResnet50-DC2/test.py]: I added a comment implicitly explaining why this needs a different setup.

Comment (kfindeisen) [reply model_packages/rbResnet50-DC2/test.py]: Sorry, which comment? I don't see it...

Comment (NimSed) [reply model_packages/rbResnet50-DC2/test.py]: In the docstrings of the classes. Or have I not pushed it?

Comment (kfindeisen) [reply model_packages/rbResnet50-DC2/test.py]: After some discussion on Slack, I finally understood that this test focuses on use of the model with `RBTransiNetInterface`, while the other focuses on its use with `NNModelPackage`. I think the parallel structure would be clearer if `TestOneCutout` were renamed to something analogous to `TestModelPackageNeighbor`; perhaps `TestTransiNetInterface`?

**Review by kfindeisen — APPROVED** [review]
I'm still not happy about leaving the tests un-run, but my only outstanding comment is about clarifying why there are two test suites.

File: model_packages/rbResnet50-DC2/test.py (hunk 5 above)
Comment (NimSed) [reply model_packages/rbResnet50-DC2/test.py]: Right. Just renamed both to something more meaningful.
